### PR TITLE
Safari/iOS focus-visible fallback

### LINKED ISSUE #344: Focus rings on mobile iOS (Safari / Chrome)
Areas this affects: 

- [ ] [Medias in the media gallery ](https://user-images.githubusercontent.com/53573494/128746651-c4db1ae5-25ce-458c-8f23-a737512aba2f.MP4)
- [ ] The stroke around the item selected in the currency & language switcher in the footer
Screen Shot 2021-08-09 at 12 55 01 PM
Screen Shot 2021-08-09 at 1 26 16 PM
- [ ] Collapsible tabs on Product Pages Screen Shot 2021-08-09 at 1 25 05 PM
- [ ] Images on product cards in product grids
- [ ] "view more information" link on Product Page's pick up availability
Image from iOS
- [ ] The "X" of the menu navigation
https://user-images.githubusercontent.com/53573494/128749490-ba4b47f5-9113-4778-a9fb-a16a438b9031.MP4

### PR DESCRIPTION
**Pull request by ludoboludo**
**Why are these changes introduced?**

Fixes #344 

**What approach did you take?**

Followed the suggested idea from Tyler.
Also kept the original styling for the `no-js` behaviour. 

**Other considerations**

There are a few things I need to look into: 
- is `Tab`, `ArrowLeft` and `ArrowRight` the only keys used to navigate around and shift focus ? I added `SPACE`, `ESC` and `ENTER` but should I also add `HOME` and `END` which seem to be used for navigating sliders 🤔 
- focus is not passed to element with the attribute `tabindex="-1"` (ex: cart notification popup, skip to product info). It most likely is more of an issue even on slower connection. So I actually added an event listener for `focus` and it seem to work now.  

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=126283513878)
- [Editor](https://os2-demo.myshopify.com/admin/themes/126283513878/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by svinkle — APPROVED** [review]

**Comment by ludoboludo** [comment]
Had to fix a rebase issue @svinkle so it dismissed your approval. Whenever you have a chance to re add it 👌

File: assets/global.js
Comment (kmeleta) [reply assets/global.js]: Perfect, that makes sense 👍

**Review by kmeleta — APPROVED** [review]
Could definitely benefit from another set of eyes, but it works well from my testing. Given the response to the experience in safari, I think it's worth the minimal js added. That said, the experience as it was didn't bother me personally.

Regarding auditing/using the existing polyfill, I think what we have here handles the cases pretty well and it's not totally clear to me what we'd be gaining from the additional lines of code.

**Review by tyleralsbury — APPROVED** [review]
This looks good to me.  I tested on Safari on my Mac - I don't have an iOS device to test with, but it should be the same behaviour.

We should log this somewhere to keep track of the fact that we do want to remove this once Safari does support `:focus-visible`.  Unfortunately, right now there's [no support on the horizon](https://caniuse.com/css-focus-visible), but I assume it will eventually be supported.

If we're not waiting for final approval from @svinkle I think we can merger this as-is and if anything comes up we can tackle it in a separate issue - it looks like this resolves the bulk of the concerns both from a UX and a11y perspective.

**Comment by ludoboludo** [comment]
Issue created: https://github.com/Shopify/dawn/issues/524

**Pull request merged**
